### PR TITLE
Refactor user invitation functionality by moving DTOs and service to Invitations namespace

### PR DESCRIPTION
**Pull request by jdelgado2002**
## Release Notes

- **New Features**
  - Introduced a new `Invitations` feature allowing users to create and manage invitations.
  - Added a user interface for sending invitations, including email input and advanced search capabilities.
  - Implemented a new controller and service for managing user invitations.

- **Improvements**
  - Enhanced localization support for invitation-related messages.
  - Implemented backend logic to handle single and multiple user invitations with error handling.
  - Added pagination and filtering capabilities for retrieving user invitations.

- **Bug Fixes**
  - Improved validation for email formats and subscription limits during invitation creation.

- **Tests**
  - Added comprehensive unit tests for the new invitation functionalities, covering various scenarios and edge cases.

### CODE REVIEW

File: aspnet-core/src/toyiyo.todo.Web.Mvc/Views/Shared/Components/_AdvancedSearch.cshtml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+                    <span class="fas fa-search" aria-hidden="true"></span>
+                </button>
+            </div>
+            <input type="search" name="Keyword" id="keyword" class="form-control txt-search" placeholder=@L("Search") />
+            <div class="input-group-append">
+                <span class="input-group-text btn-clear" id="btnClearSearch">
+                    <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-x-circle-fill" viewBox="0 0 16 16">
+                        <path d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0M5.354 4.646a.5.5 0 1 0-.708.708L7.293 8l-2.647 2.646a.5.5 0 0 0 .708.708L8 8.707l2.646 2.647a.5.5 0 0 0 .708-.708L8.707 8l2.647-2.646a.5.5 0 0 0-.708-.708L8 7.293z"></path>
+                    </svg>
+                </span>
```
[the remark this answers is not in the document]
Comment (jdelgado2002) [reply aspnet-core/src/toyiyo.todo.Web.Mvc/Views/Shared/Components/_AdvancedSearch.cshtml:18]: button messes up the display

File: aspnet-core/src/toyiyo.todo.Core/Invitations/UserInvitationManager.cs [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        }
+        private async Task SendInvitationEmailAsync(UserInvitation invitation)
+        {
+            //todo: setup email sending
+            var link = $"https://yourapp.com/Account/Register?token={invitation.Token}";
+            var subject = "You are invited!";
+            var body = $"Please click the following link to register: {link}";
+            //todo: setup email sending
+            return;//_emailSender.SendAsync(invitation.Email, subject, body);
+        }
```
[the remark this answers is not in the document]
Comment (jdelgado2002) [reply aspnet-core/src/toyiyo.todo.Core/Invitations/UserInvitationManager.cs:197]: WE'll implement this in a future PR
